frontend/app/routes/$lang+/_public+/apply+/$id+/federal-provincial-territorial-benefits.tsx

```diff
@@ -68,6 +69,11 @@ export async function action({ request, params }: ActionFunctionArgs) {
         message: 'provincial-program',
         path: ['provincialTerritorialSocialProgram'],
       });
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: 'empty-province',
+        path: ['province'],
+      });
```

Comment:
This newly added validation doesn't take into account when a province has been selected.

When a user selects "Yes, I have provincial or territorial benefits.", then selects any province, then leaves the Provincial social program question (radio buttons) unanswered, the "Province must be selected" error appears even though we selected a province:

![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/55502055/fbada3da-3f8b-452a-9d06-148876dd3336)